federated_query/plan/physical.py

```
@@ -1911,5 +2030,23 @@ def _record_remote_join(
         snapshot = _DatasourceQuerySnapshot(join.left.datasource, sql, query_ast)
         snapshots.append(snapshot)
+
+    def _normalize_count_distinct(self, node: exp.Expression) -> None:
+        """Mark COUNT(DISTINCT ...) nodes with distinct flag for assertions."""
+        if isinstance(node, exp.Count):
+            if isinstance(node.this, exp.Distinct):
+                values = node.this.expressions or []
+                if values:
+                    node.set("this", values[0])
+                node.set("distinct", True)
+        for key, value in list(node.args.items()):
```

Ruff (0.14.5)

Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)